chips/apollo3/src/iom.rs

```diff
@@ -605,13 +612,68 @@ impl<'a> Iom<'_> {
                 }
 
                 self.spi_read_buffer.replace(buf);
+
+                if self.read_len.get() > self.read_index.get() {
+                    let remaining_bytes = (self.read_len.get() - self.read_index.get()).min(32);
+                    if remaining_bytes > 4 {
+                        self.registers
+                            .fifothr
+                            .modify(FIFOTHR::FIFORTHR.val(remaining_bytes as u32));
+                    } else {
+                        self.registers.fifothr.modify(FIFOTHR::FIFORTHR.val(1));
+                    }
```

Comment:
Why the `> 4` here? I don't see anything special about the 2 or 3 byte case in the documentation?
<img width="796" alt="image" src="https://github.com/tock/tock/assets/339422/d0408d8f-a3c8-46d2-9c87-3e96821a4e1f">

If this is needed for some reason, can you add a comment explaining?

Reply:
I just copied it from the original I2C implementation. I don't think it's required though, so I've dropped it